enterprise/utils.py

```diff
@@ -49,6 +49,39 @@ def __init__(self, *args, **kwargs):
         super(NotConnectedToEdX, self).__init__(*args, **kwargs)
 
 
+class ConditionalEmailConnection(object):
+    """
+    Conditionally open an email connection as a context manager.
+    """
+
+    def __init__(self, open_conn=False):
+        """
+        Save the preference of whether to open a connection.
+
+        Args:
+            open_conn (bool): Whether the context manager should create and open an email
+            connection when entering a `with` statement.
+        """
+        self.open_conn = open_conn
+        self.conn = None
+
+    def __enter__(self):
+        """
+        Create and open the mail connection if necessary.
+        """
+        if self.open_conn:
+            self.conn = mail.get_connection()
+            self.conn.open()
+        return self.conn
+
+    def __exit__(self, *excinfo):
+        """
+        Close the email connection if it was opened.
+        """
+        if self.conn and self.open_conn:
```

Comment:
This is a nit, but it's better just to check `if self.conn` here - regardless of whether a connection was supposed to be open or not, if the connection was opened, it should be closed.